pset/updater_test.go

```diff
@@ -0,0 +1,71 @@
+package pset
+
+import (
+	"encoding/hex"
+	"github.com/stretchr/testify/assert"
+	"github.com/vulpemventures/go-elements/transaction"
+	"testing"
+)
+
+func TestUpdaterAddInput(t *testing.T) {
+	inputs := make([]*transaction.TxInput, 0)
+	output := make([]*transaction.TxOutput, 0)
+	pset, err := New(inputs, output, 2, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	updater, err := NewUpdater(pset)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	hash, err := hex.DecodeString(
+		"000102030405060708090a0b0c0d0e0f000102030405060708090a0b0c0d0e0f")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	txInput := transaction.TxInput{
+		Hash:  hash,
+		Index: 2,
+	}
+	updater.AddInput(&txInput)
```

Comment:
We should assert that, before this function is called, the transaction has 0 inputs and 0 outputs

Reply:
Done